### DISCUSSION TITLE
Is it possible to make text in an Autocomplete component wrap?

### DISCUSSION THREAD

**Discussion by eagerestwolf** [skeletonlabs/skeleton#2091] [Q&A] [answered]
So, I am working on an open-source Transportation Management System using Skeleton. Progress is slow (as major apps tend to be), however I have gotten as far as making a Google Maps search component. Yes, I could have just used the default autocomplete widget (but ugly), and I know I could have done some PostCSS `@apply` magic, but I like this approach better since I have full control over the styling and the interactivity. I will say kudos to the devs, this probably wasn't intentional, but if you change the `AutocompleteOption`s dynamically, the component doesn't freak out, it just works. Now, since I am working with Google Maps stuff, I obviously don't know how long the strings I am displaying are, so I just set the autocomplete width to `w-modal` (probably a very naive approach I know) because I want to prevent the component from being wider than the display, but sometimes the text overflows outside of the popup. Is there a way to make the text wrap, like using `flex-auto` or something? I have included the source of my component below, if that helps.

> Side note: I know I shouldn't be using `console.error()` so liberally, but I am in **_VERY_** early development at the moment (as in this project doesn't even have a repository yet), so I am the only one who will see those messages. Later, those will be replaced with proper error handling.

**UPDATE**: I removed the `w-modal` from the popup, and I have to say I like that better. The popup now dynamically scales to the width of the content. Also, tried `flex-auto` in the `regionItem` prop to no avail. Small screens are my only real issue here. If you're trying the code for yourself, searching for "united denver" should bring up the United States Mint Gift Shop in Denver, CO which is sufficiently long to cause an overflow on smaller screens.

```svelte
<script lang="ts">
  import type { Loader } from "@googlemaps/js-api-loader";
  import {
    Autocomplete,
    type AutocompleteOption,
    popup,
    type PopupSettings
  } from "@skeletonlabs/skeleton";
  import { onMount } from "svelte";

  import { getLoader } from "$lib/client/googleMapsApiLoader";

  let inputPlace: string = "";
  let loader: Loader;
  let placeOptions: AutocompleteOption[] = [];
  export let value: google.maps.GeocoderResult | undefined;
  const popupSettings: PopupSettings = {
    event: "focus-click",
    placement: "top",
    target: "popupAutocomplete"
  };

  onMount((): void => {
    loader = getLoader();
  });

  function onSearchInput(): void {
    loader
      .importLibrary("places")
      .then((places) => {
        const autocompleteService = new places.AutocompleteService();

        autocompleteService
          .getPlacePredictions({ input: inputPlace })
          .then((res) => {
            placeOptions = [];

            if (res.predictions.length > 0) {
              res.predictions.forEach((prediction) => {
                placeOptions.push({
                  label: `<b>${prediction.structured_formatting.main_text}</b>, ${prediction.structured_formatting.secondary_text}`,
                  meta: {
                    prediction
                  },
                  value: prediction.place_id
                });
              });
            }
          })
          .catch((err) => {
            console.error(err);
          });
      })
      .catch((err) => {
        console.error(err);
      });
  }

  function onPlaceSelect(event: CustomEvent<AutocompleteOption>): void {
    loader
      .importLibrary("geocoding")
      .then((geocoding) => {
        const geocoder = new geocoding.Geocoder();

        geocoder
          .geocode({ placeId: event.detail.meta.prediction.place_id })
          .then((res) => {
            const [result] = res.results;

            value = result;
            inputPlace = result.formatted_address;
          })
          .catch((err) => {
            console.error(err);
          });
      })
      .catch((err) => {
        console.error(err);
      });
  }
</script>

<input
  class="autocomplete input"
  type="search"
  name="autocomplete-search"
  on:input={onSearchInput}
  bind:value={inputPlace}
  placeholder="Search..."
  use:popup={popupSettings} />
<div class="card w-modal p-4 shadow-xl" data-popup="popupAutocomplete">
  <Autocomplete bind:input={inputPlace} options={placeOptions} on:selection={onPlaceSelect} />
  <div class="bg-surface-100-800-token arrow" />
</div>

```

**Comment by SebasF1349** [comment, marked as answer]
This is more of a tailwind/css issue. The Autocomplete component renders each option as a `button` inside a `li`, as can be seen in the source code: [https://github.com/skeletonlabs/skeleton/blob/master/packages/skeleton/src/lib/components/Autocomplete/Autocomplete.svelte#L164C43-L164C56](https://github.com/skeletonlabs/skeleton/blob/master/packages/skeleton/src/lib/components/Autocomplete/Autocomplete.svelte#L164C43-L164C56)

So you need to use the `regionButton` prop to add classes to it. In this case you need to use the whitespace property [https://tailwindcss.com/docs/whitespace](https://tailwindcss.com/docs/whitespace). Adding `regionButton="!whitespace-normal"` to the `Autocomplete` component should work. Here is a working example: [https://stackblitz.com/edit/skeletonlabs-repl-jhoxcz?file=src%2Froutes%2F%2Bpage.svelte](https://stackblitz.com/edit/skeletonlabs-repl-jhoxcz?file=src%2Froutes%2F%2Bpage.svelte)

Note that the bang `!` is important to overwrite the `whitespace-nowrap` added by default by Skeleton [https://github.com/skeletonlabs/skeleton/blob/master/packages/plugin/src/styles/components/lists.css#L39](https://github.com/skeletonlabs/skeleton/blob/master/packages/plugin/src/styles/components/lists.css#L39)

You also may want to play with word break to handle very long words [https://tailwindcss.com/docs/word-break](https://tailwindcss.com/docs/word-break)

**Reply by eagerestwolf** [reply]
Yeah, sorry about that. I didn't even think of looking at the source. That did the trick though. I just had to make a couple of small changes to get it just right. I also added a `text-left` to the `regionButton` as well because it messed up the spacing. I also had to wrap the whole result entry in a `<span>` tag because the `whitespace-normal` was wrapping the `<b>` tag separately. Minor issues I can work with though. I'm still learning a lot of the nuances of working with Tailwind, but I'm definitely getting there.

**Reply by SebasF1349** [reply]
No problem, happy to help.
